pkg/buses/references.go

```diff
@@ -16,22 +16,86 @@
 
 package buses
 
-import "fmt"
+import (
+	"fmt"
 
+	channelsv1alpha1 "github.com/knative/eventing/pkg/apis/channels/v1alpha1"
+)
+
+// BusReference references a Bus or ClusterBus within the cluster by name and
+// namespace. For ClusterBus the namespace will be an empty string.
 type BusReference struct {
 	Name      string
 	Namespace string
 }
 
+// NewBusReferenceFromNames creates a BusReference for a name and namespace.
+func NewBusReferenceFromNames(name, namespace string) BusReference {
+	return BusReference{
+		Name:      name,
+		Namespace: namespace,
+	}
+}
+
 func (r *BusReference) String() string {
-	return fmt.Sprintf("%s/%s", r.Namespace, r.Name)
+	if len(r.Namespace) != 0 {
```

Comment:
I have a slight preference for `r.Namespace != ""`; it makes bugs like #377 slightly easier to catch, IMO.

Reply:
done